lib/icingadb/icingadb-objects.cpp

```diff
@@ -2642,8 +2642,8 @@ Dictionary::Ptr IcingaDB::SerializeState(const Checkable::Ptr& checkable)
 
 		if (!cr->GetCommand().IsEmpty())
 			attrs->Set("check_commandline", FormatCommandLine(cr->GetCommand()));
-		attrs->Set("execution_time", TimestampToMilliseconds(fmax(0.0, cr->CalculateExecutionTime())));
-		attrs->Set("latency", TimestampToMilliseconds(cr->CalculateLatency()));
+		attrs->Set("execution_time", fmin(pow(2.0, 32.0) - 1.0, TimestampToMilliseconds(fmax(0.0, cr->CalculateExecutionTime()))));
+		attrs->Set("latency", fmin(pow(2.0, 32.0) - 1.0, TimestampToMilliseconds(cr->CalculateLatency())));
```

Comment:
Why do you use `fmin()` and `pow()` here? `TimestampToMilliseconds()` returns `long long` already, so shouldn't simply `std::min(UINT32_MAX, ...)` do the trick here?